### PR TITLE
Order customer info redesign

### PR DESCRIPTION
**Pull request by mirekm**
Both order and draft update to customer details section.

### **Order** new/old:
image

image

### **Draft** new/old:
image
image

### CODE REVIEW

File: src/orders/components/OrderDetailsPage/Title.tsx [hunk 1]
```diff
@@ -29,7 +29,12 @@ const Title = (props: TitleProps) => {
   const { order } = props;
 
   if (!order) {
-    return null;
+    return (
+      <div className={classes.container}>
+        <Skeleton style={{ width: "8em" }} />
+        <Skeleton style={{ width: "10em" }} />
```
Comment (lkostrowski) [inline src/orders/components/OrderDetailsPage/Title.tsx]: you can do `width={8}` and if you want t obe precise `__width="8em"`

**Comment by mirekm** [comment]
Thanks for the review, @lkostrowski 
I will merge it as soon as the https://github.com/saleor/saleor-dashboard/pull/6150 is cleared (as it was branched out from it)

**Comment by lkostrowski** [comment]
withdrawn my review - once #6150 is merged, lets rebase this one and target main directly

File: src/orders/components/OrderCustomer/OrderCustomer.test.tsx [hunk 2]
```diff
@@ -0,0 +1,248 @@
...
+jest.mock("@dashboard/auth/hooks/useUserPermissions");
+jest.mock("./CustomerEditForm", () => ({
+  CustomerEditForm: () => <div data-test-id="customer-edit-form">CustomerEditForm</div>,
+}));
+jest.mock("./PickupAnnotation", () => ({
+  PickupAnnotation: () => <div>PickupAnnotation</div>,
+}));
+jest.mock("./AddressTextError", () => ({
+  AddressTextError: () => <div>AddressTextError</div>,
+}));
```
Comment (lkostrowski) [inline src/orders/components/OrderCustomer/OrderCustomer.test.tsx:24]: not sure about value of these tests, where contents are mocked? I think we should focus on these testing abstractions
1. Units, isolated from UI. Utils, models, mapping logic, minimum react and markup, maximum of business logic, data transformations etc
2. Integration logic - render full markup, test if everything renders together. Less edge cases ,more integration flow
